src/aco/ant.rs

```diff
@@ -48,28 +75,7 @@ impl<R: Rng> CanonicalAnt<R> {
 }
 
 impl<R: Rng> Ant for CanonicalAnt<R> {
-  /// Clears iteration specific data like visited vertices or path.
-  fn clear(&mut self) {
-    for i in 0..self.solution_size {
-      self.unvisited.push(i);
-    }
-    self.path.clear();
-    self.stuck = false;
-  }
-  /// Returns vector of vertices in order of visiting
-  fn get_path(&self) -> &[usize] {
-    &self.path
-  }
-  /// Selects an vertex to start from
-  fn chose_staring_place(&mut self) {
-    let start: usize = self.rng.gen_range(0..self.solution_size);
-    self.unvisited.remove(&start);
-    self.path.push(start);
-  }
-  /// Returns true when there is no valid next vertex with path not fully constructed.
-  fn is_stuck(&self) -> bool {
-    self.stuck
-  }
+  standard_ant_impl!();
```

Comment:
What is the reason to use a macro here? I do not see this code being used in multiple places moreover there are no parameters provided.

Reply:
Ok I see this being used below another time. Hmm, but I do not know whether using a macro is the "go to way" for resolving code repetition in Rust.

Reply:
Ok, I've done some research and it seems there is no better way unfortunately :D 

GJ then 🎉